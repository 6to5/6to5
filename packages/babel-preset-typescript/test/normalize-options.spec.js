import normalizeOptions from "../src/normalize-options";
describe("normalize options", () => {
  describe("Babel_8_breaking", () => {
    let old_babel_8_breaking_env;
    beforeAll(() => {
      old_babel_8_breaking_env = process.env.BABEL_8_BREAKING;
      process.env.BABEL_8_BREAKING = "1";
    });
    afterAll(() => {
      if (old_babel_8_breaking_env) {
        process.env.BABEL_8_BREAKING = old_babel_8_breaking_env;
      } else {
        delete process.env.BABEL_8_BREAKING;
      }
    });
    it("should throw on unknown options", () => {
      expect(() => normalizeOptions({ allowNamespace: true })).toThrowError(
        "@babel/preset-typescript: 'allowNamespace' is not a valid top-level option.\n- Did you mean 'allowNamespaces'?",
      );
    });
    it.each([
      "allowDeclareFields",
      "allExtensions",
      "allowNamespaces",
      "isTSX",
      "onlyRemoveTypeImports",
    ])("should throw when `%p` is not a boolean", optionName => {
      expect(() =>
        normalizeOptions({ [optionName]: 0 }),
      ).toThrowErrorMatchingSnapshot();
    });
    it.each(["jsxPragma", "jsxPragmaFrag"])(
      "should throw when `%p` is not a string",
      optionName => {
        expect(() =>
          normalizeOptions({ [optionName]: 0 }),
        ).toThrowErrorMatchingSnapshot();
      },
    );
    it("should not throw when options is not defined", () => {
      expect(() => normalizeOptions()).not.toThrowError();
    });
    it("default values", () => {
      expect(normalizeOptions({})).toMatchInlineSnapshot(`
        Object {
          "allExtensions": false,
          "allowDeclareFields": true,
          "allowNamespaces": true,
          "isTSX": false,
          "jsxPragma": "React",
          "jsxPragmaFrag": "React.Fragment",
          "onlyRemoveTypeImports": true,
        }
      `);
    });
  });
  (process.env.BABEL_8_BREAKING ? describe.skip : describe)("Babel 7", () => {
    it("should not throw on unknown options", () => {
      expect(() =>
        normalizeOptions({ allowNamespace: true }),
      ).not.toThrowError();
    });
    it.each(["allowDeclareFields", "allowNamespaces", "onlyRemoveTypeImports"])(
      "should not throw when `%p` is not a boolean",
      optionName => {
        expect(() => normalizeOptions({ [optionName]: 0 })).not.toThrowError();
      },
    );
    it.each(["jsxPragma"])(
      "should throw when `%p` is not a string",
      optionName => {
        expect(() => normalizeOptions({ [optionName]: 0 })).not.toThrowError();
      },
    );
    it.each(["allExtensions", "isTSX"])(
      "should throw when `%p` is not a boolean",
      optionName => {
        expect(() =>
          normalizeOptions({ [optionName]: 0 }),
        ).toThrowErrorMatchingSnapshot();
      },
    );
    it.each(["jsxPragmaFrag"])(
      "should throw when `%p` is not a string",
      optionName => {
        expect(() =>
          normalizeOptions({ [optionName]: 0 }),
        ).toThrowErrorMatchingSnapshot();
      },
    );
    it("default values", () => {
      expect(normalizeOptions({})).toMatchInlineSnapshot(`
        Object {
          "allExtensions": false,
          "allowDeclareFields": undefined,
          "allowNamespaces": undefined,
          "isTSX": false,
          "jsxPragma": undefined,
          "jsxPragmaFrag": "React.Fragment",
          "onlyRemoveTypeImports": undefined,
        }
      `);
    });
  });
});
