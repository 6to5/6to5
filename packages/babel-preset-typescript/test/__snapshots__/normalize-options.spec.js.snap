// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalize options Babel 7 should throw when \`"allExtensions"\` is not a boolean 1`] = `"@babel/preset-typescript: 'allExtensions' option must be a boolean."`;

exports[`normalize options Babel 7 should throw when \`"isTSX"\` is not a boolean 1`] = `"@babel/preset-typescript: 'isTSX' option must be a boolean."`;

exports[`normalize options Babel 7 should throw when \`"jsxPragmaFrag"\` is not a string 1`] = `"@babel/preset-typescript: 'jsxPragmaFrag' option must be a string."`;

exports[`normalize options Babel_8_breaking should throw when \`"allExtensions"\` is not a boolean 1`] = `"@babel/preset-typescript: 'allExtensions' option must be a boolean."`;

exports[`normalize options Babel_8_breaking should throw when \`"allowDeclareFields"\` is not a boolean 1`] = `"@babel/preset-typescript: 'allowDeclareFields' option must be a boolean."`;

exports[`normalize options Babel_8_breaking should throw when \`"allowNamespaces"\` is not a boolean 1`] = `"@babel/preset-typescript: 'allowNamespaces' option must be a boolean."`;

exports[`normalize options Babel_8_breaking should throw when \`"isTSX"\` is not a boolean 1`] = `"@babel/preset-typescript: 'isTSX' option must be a boolean."`;

exports[`normalize options Babel_8_breaking should throw when \`"jsxPragma"\` is not a string 1`] = `"@babel/preset-typescript: 'jsxPragma' option must be a string."`;

exports[`normalize options Babel_8_breaking should throw when \`"jsxPragmaFrag"\` is not a string 1`] = `"@babel/preset-typescript: 'jsxPragmaFrag' option must be a string."`;

exports[`normalize options Babel_8_breaking should throw when \`"onlyRemoveTypeImports"\` is not a boolean 1`] = `"@babel/preset-typescript: 'onlyRemoveTypeImports' option must be a boolean."`;
